### ISSUE TITLE
Keras Metrics not updating when passed as objects

### ISSUE THREAD

**Issue by makisgrammenos** [keras-team/keras#16484]
Hello,

I am experiencing issues with keras metrics  while trying to train  the following model.
```
 inputs=layers.Input(shape=(128,128,64,1))
    x =layers.Conv3D(filters=16,kernel_size=(3,3,3),activation='relu',padding='same')(inputs)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
    x = layers.BatchNormalization()(x)
    x = layers.Conv3D(filters=32,kernel_size=(3,3,3),activation='relu',padding='same')(x)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
    x = layers.BatchNormalization()(x)
    x = layers.Conv3D(filters=64,kernel_size=(3,3,3),activation='relu',padding='same')(x)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
    x = layers.BatchNormalization()(x)
    x = layers.Conv3D(filters=128,kernel_size=(2,2,2),activation='relu',padding='same')(x)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
    x = layers.BatchNormalization()(x)

    x = layers.Conv3D(filters=128,kernel_size=(2,2,2),activation='relu',padding='same')(x)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
    x = layers.BatchNormalization()(x)

    
    x = layers.Conv3D(filters=128,kernel_size=(2,2,2),padding='same',activation='relu')(x)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
    
    
    x = layers.Conv3D(filters=128,kernel_size=(2,2,2),padding='same',activation='relu')(x)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
    
    x = layers.Conv3D(filters=128,kernel_size=(2,2,2),padding='same',activation='relu')(x)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
    
    x = layers.Conv3D(filters=128,kernel_size=(2,2,2),padding='same',activation='relu')(x)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
    
    
    x = layers.Conv3D(filters=256,kernel_size=(2,2,2),padding='same',activation='relu')(x)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
    
    x = layers.Conv3D(filters=256,kernel_size=(2,2,2),padding='same',activation='relu')(x)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
   
    
    
    x = layers.Conv3D(filters=256,kernel_size=(2,2,2),padding='same',activation='relu')(x)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
    x = layers.Conv3D(filters=256,kernel_size=(2,2,2),padding='same',activation='relu')(x)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
    x = layers.Conv3D(filters=256,kernel_size=(2,2,2),padding='same',activation='relu')(x)
    x = layers.MaxPooling3D(pool_size=(2,2,2),padding='same')(x)
    
    x = layers.Flatten()(x)
    x = layers.Dense(units=256,activation='relu')(x)

    x = layers.Dense(units=256,activation='relu')(x)

    x = layers.Dense(units=256,activation='relu')(x)
    outputs = layers.Dense(4,activation='softmax')(x)
    return tf.keras.Model(inputs=inputs,outputs=outputs)


```
If I pass the metrics objects in ` model.compile` like this:

``` model.compile(optimizer=optmizer,loss='categorical_crossentropy',metrics=[tf.keras.metrics.Accuracy(),tf.keras.metrics.Precision()])```
the metrics are not updating.They always stay at 0.0000e+00.
I only managed to get accuracy to work by passing it like this

``` model.compile(optimizer=optmizer,loss='categorical_crossentropy',metrics=['acurracy',tf.keras.metrics.Precision()])```.
I am using  `Ubuntu 21.10` with `python 3.8.8` and `tensorflow 2.8.0`
Also I am using a custom data generator

**Comment by sushreebarsa** [comment]
@makisgrammenos 
In order to expedite the trouble-shooting process, please provide a complete code snippet to reproduce the issue reported here. Thanks!
